tests/FakeIngest.php

```diff
@@ -40,14 +40,18 @@ public function assertWrittenTimes(int $expected): self
         return $this;
     }
 
-    public function assertLatestWrite(string|array $key, mixed $expected = null): self
+    public function assertWrite(int $index, string|array|Closure $key, mixed $expected = null): self
```

Comment:
This new method allows us to assert against any particular write, not just the latest.

The `assertLatestWrite` method still exists and is what we will use in 99% of cases.